staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -498,7 +499,13 @@ func (s sortableStoreElements) Swap(i, j int) {
 // WaitUntilFreshAndList returns list of pointers to `storeElement` objects along
 // with their ResourceVersion and the name of the index, if any, that was used.
 func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion uint64, matchValues []storage.MatchValue) (result []interface{}, rv uint64, index string, err error) {
-	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) && w.notFresh(resourceVersion) {
+	var shouldRegister bool
+	shouldRegister, err = w.shouldRegisterAsWaitingForFresh(ctx, resourceVersion)
+	if err != nil {
```

Comment:
I started wondering about the syntax of `Supports` method. Basically, the only case when we return an error is if we don't recognize the requested feature.
That effectively is not an error - it's rather a "programmer bug" to call it for something we don't know.
In addition, there are places where we just ignore this errors.

I would actually suggest changing the syntax of `Supports` to return just a bool, and for unrecognized features just log an error and return false. 
I basically can't imagine a reasonable error handling for those cases.

Reply:
Done